src/components/Header.jsx

```diff
@@ -1,20 +1,39 @@
-import React from 'react';
-import { Navbar, Nav } from 'react-bootstrap';
-import { NavLink } from 'react-router-dom';
+import React from "react";
+import { Navbar, Nav } from "react-bootstrap";
+import { NavLink } from "react-router-dom";
+import AuthenticationModal from './AuthenticationModal';
+import store from '../store/store';
+import { LOGIN, REGISTER} from '../constants/modal-action-types';
 
-const Header = () => (
-  <Navbar expand="lg" className="navbar-custom">
-    <NavLink exact to="/" className="brand">
-        Author&apos;s Haven
-    </NavLink>
-    <Navbar.Toggle aria-controls="basic-navbar-nav" />
-    <Navbar.Collapse id="basic-navbar-nav">
-      <Nav className="ml-auto">
-        <NavLink to="/login" className="nav-link">
-            Login
+export class Header extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalShow: false,
+      isRegister: false
+    };
+  }
+
+  render() {
+    return (
+      <Navbar expand="lg" className="navbar-custom">
+        <NavLink exact to="/" className="brand">
+          Author&apos;s Haven
         </NavLink>
-      </Nav>
-    </Navbar.Collapse>
-  </Navbar>
-);
+        <Navbar.Toggle aria-controls="basic-navbar-nav" />
+        <Navbar.Collapse id="basic-navbar-nav">
+          <Nav className="ml-auto">
+            <Nav.Item onClick={() => store.dispatch({type: LOGIN, payload: null})} className="nav-link auth-btn ">
+              Login
+            </Nav.Item>
+            <Nav.Item onClick={() => store.dispatch({type: REGISTER, payload: null})} className="nav-link auth-btn get-started">
```

Comment:
No need of passing a payload of null, just pass the type only and exclude payload altogether, ie `dispatch({ type: REGISTER })`